runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -4146,7 +4146,7 @@ TR::Register *J9::Power::TreeEvaluator::VMifInstanceOfEvaluator(TR::Node *node,
    uint8_t num_PICS = TR::TreeEvaluator::interpreterProfilingInstanceOfOrCheckCastInfo(cg, instanceOfNode, guessClassArray);
 
    bool testEqualClass = instanceOfOrCheckCastNeedEqualityTest(instanceOfNode, cg);
-   bool testCastClassIsSuper = instanceOfOrCheckCastNeedSuperTest(instanceOfNode, cg);
+   bool testCastClassIsSuper = castClassAddr && instanceOfOrCheckCastNeedSuperTest(instanceOfNode, cg);
```

Comment:
Just curious, what's the motivation for this part?

Reply:
This affects the flag two lines below:
```
bool needsHelperCall = needHelperCall(testCastClassIsSuper, isFinalClass);
// !testCastClassIsSuper && !isFinalClass
```
Issue is in cases where cast class is unknown at compile time, `testCastClassIsSuper` could still be true which end up inlining classTest and superClassTest which fails if it was an interface. By falsifying `testCastClassIsSuper` when class is unknown, this allows `needsHelperCall` to be true and the code generated is inlining the classTest (for the fast path) and a call to the helper (for the slow path).

The planned improvement work would have a more elaborate inlined fast path.

Reply:
OK, so basically this is an optimization and not needed for correctness right? If we don't have a `castClassAddr` we'll avoid generating an inline superclass test (which we know will fail).

If that's a case can you add a comment to that effect? It will help us remember that this optimization exists.

Reply:
This is for correctness, if we don't have `castClassAddr` we will generate superclass test instead of callHelper (which will fail if it's an interface).

Added the comment.